miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/file/MiniAppFilePicker.kt

```diff
@@ -0,0 +1,20 @@
+package com.rakuten.tech.mobile.miniapp.file
+
+import android.net.Uri
+import android.webkit.ValueCallback
+
+/** The file picker of a miniapp. **/
+interface MiniAppFilePicker {
+
+    /**
+     * Request to pick a file from the device within the miniapp webview per [requestCode].
+     * @param filePathCallback a file path callback to pass to HostApp has been found
+     * by [MiniAppWebChromeClient.onShowFileChooser].
+     * @param callback to pass the request code of file picking using an intent inside sdk,
+     * which will also be used to retrieve the data using onActivityResult in the host app.
+     */
+    fun requestFile(
```

Comment:
Related to my comment above about simplifying the integration: I don't think we need this method.

However, it would be nice if the Host App had the ability to override the `onShowFileChooser` functionality completely, so maybe we could add that here? Basically refactor your implementation from `MiniAppWebChromeClient` into this class (and then the Host App can choose to override that implementation if they want to).

```
open class MiniAppFilePicker(var requestCode: Int) {

    internal var callback: ValueCallback<Array<Uri>>? = null

    fun onShowFileChooser(
        filePathCallback: ValueCallback<Array<Uri>>?,
        fileChooserParams: WebChromeClient.FileChooserParams?,
        context: Context
    ) : Boolean {
        try {
            this.callback = filePathCallback
            val intent = fileChooserParams?.createIntent()
            (context as Activity).startActivityForResult(intent, requestCode)
        } catch (e: Exception) {
            throw (MiniAppFilePickingException(e.message.toString()))
        }
        return true
    }

    fun onReceivedFiles(files: Array<Uri>) {
        callback?.onReceiveValue(files)
    }
}
```

Reply:
Yeah, your idea is working well, specially `onReceivedFiles` is convenient for the HostApp, so I have updated it.